mmedit/models/backbones/sr_backbones/iconvsr.py

```diff
@@ -0,0 +1,362 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from mmcv.cnn import ConvModule
+from mmcv.runner import load_checkpoint
+
+from mmedit.models.common import (PixelShufflePack, ResidualBlockNoBN,
+                                  flow_warp, make_layer)
+from mmedit.models.registry import BACKBONES
+from mmedit.utils import get_root_logger
+from .basicvsr import ResidualBlocksWithInputConv, SPyNet
+from .edvr_net import PCDAlignment, TSAFusion
+
+
+@BACKBONES.register_module()
+class IconVSR(nn.Module):
+    """IconVSR network structure for video super-resolution.
+
+    Support only x4 upsampling.
+    Paper:
+        BasicVSR: The Search for Essential Components in Video Super-Resolution
+        and Beyond, arXiv, 2020
+
+    Args:
+        mid_channels (int): Channel number of the intermediate features.
+            Default: 64.
+        num_blocks (int): Number of residual blocks in each propagation branch.
+            Default: 30.
+        keyframe_stride (int): Number determining the keyframes. If stride=5,
+            then the (0, 5, 10, 15, ...)-th frame will be the keyframes.
+            Default: 5.
+        temporal_padding (int): Number of frames to be padded at two ends of
+            the sequence. 2 for REDS and 3 for Vimeo-90K. Default: 2
+        spynet_pretrained (str): Pre-trained model path of SPyNet.
+            Default: None.
+        edvr_pretrained (str): Pre-trained model path of EDVR (for refill).
+            Default: None.
+    """
+
+    def __init__(self,
```

Comment:
Unit test for this module is missing.

Reply:
The unit test for this module is in `tests/test_iconvsr.py`.